### PR TITLE
Trigger cloud intergration workflow for PRs and pushes against master

### PR DESCRIPTION
**Pull request by dimko**
Run cloud e2e tests when pushing to master against **master**, not **main**, as it previously was.

### CODE REVIEW

**Review by kaskavel — APPROVED** [review]

**Comment by dimko** [comment]
@kaskavel @Ferroin I'd appreciate if you'd take a look. I changed the event type from pull_request to pull_request_target, so the context now is that of the base, not that of the PR, thus enabling us (hopefully) to run regression when issuing PRs in this repo. Since we're now using the context of the base repo (netdata/netdata) and now the fork, external changes to the workflow won't trigger its execution and also once this workflow get's triggered it will have access to repo's secrets. Btw, we might want to limit this to specific type of changes (any suggestions?), like changes to the C code only that affect agent code compilation and functionality. WDYT?

**Comment by dimko** [comment]
Maybe use the following paths, just like we do in the tests workflow ?

`paths:
      - 'CMakeLists.txt'
      - '**.c'
      - '**.h'`

**Review by kaskavel — APPROVED** [review]

File: .github/workflows/cloud_regression.yml [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+    paths:
+      - 'CMakeLists.txt'
+      - '**.c'
+      - '**.h'
+  pull_request_target:
+    branches: [master]
+    paths:
+      - 'CMakeLists.txt'
+      - '**.c'
+      - '**.h'
```
Comment (Ferroin) [inline .github/workflows/cloud_regression.yml:14]: Unless our goal is solely to test cloud integration, we probably want a larger list here.

Even if our goal is _just_ testing cloud integration though, we also should be triggering on any changes to the various ACLK related submodules (such as the `aclk/aclk-schemas` and `mqtt_websockets`), which will not match this.

Comment (dimko) [reply .github/workflows/cloud_regression.yml:14]: Thanks @Ferroin will add these as well. Is it just these 2? Are there any other?

Comment (Ferroin) [reply .github/workflows/cloud_regression.yml:14]: I _think_ there are others, but I’m not 100% certain. @underhood is probably the person to ask.

Comment (underhood) [reply .github/workflows/cloud_regression.yml:14]: ACLK related submodules... there are few of them but all under `mqtt_websockets` folder (it is a tree). For changes on those however we are interested whether the connection to cloud can be made and maintained. No need to go trough whole test suite.
Change to `mqtt_websockets` and its children may affect ability to maintain or establish connection. They will not affect any application layer messages or logic.

Comment (Ferroin) [reply .github/workflows/cloud_regression.yml:14]: Given that this is supposed to be triggering regression testing, I think we (probably) do want to run for changes there. A broken transport layer is still a regression, even if it’s not necessarily in the core agent code.

We definitely need to retest on changes to `aclk/aclk-schemas` though, that’s application-layer stuff.

Comment (dimko) [reply .github/workflows/cloud_regression.yml:14]: Thanks guys, so to make sure I understood correctly and apply the required changes. 

I should edit the list of tracked files to the following:


paths:
      - 'CMakeLists.txt'
      - '**.c'
      - '**.h'
      -'aclk/aclk-schemas/*'
      -'mqtt_websockets/*'

**Review by Ferroin — APPROVED** [review]

**Comment by dimko** [comment]
Adding **.cc** and **.cpp** extensions to the list of tracked files

**Review by Ferroin — APPROVED** [review]

**Review by stelfrag — APPROVED** [review]

**Pull request merged**